django_embed_template/templatetags/embed.py

```diff
@@ -1,28 +1,43 @@
+import copy
+
 from django.template.library import Library
 from django.template.base import TextNode
 from django.template.loader_tags import (
-    BLOCK_CONTEXT_KEY, do_include, ExtendsNode, BlockContext, BlockNode
+    BLOCK_CONTEXT_KEY, do_include, ExtendsNode, BlockContext, BlockNode, Node
 )
 from django.utils import six
 
 register = Library()
 
 
-class EmbedNode(ExtendsNode):
+class EmbedNode(Node):
+    """
+    Example:
+        {% embed 'template.html' with var_name="var_value" %}
+            {% block template_content %}
+                my content
+            {% endblock %}
+        {% endembed %}
+    """
     must_be_first = False
     context_key = 'embeds_context'
 
     def __init__(self, nodelist, include_node, template_dirs=None):
-        super().__init__(nodelist, include_node.template, template_dirs)
+        super(EmbedNode, self).__init__()
+
         self.include_node = include_node
+        self.extend_node = ExtendsNode(nodelist, include_node.template, template_dirs)
 
     def render(self, context):
-        old_embeds_context = context.render_context.setdefault(
+        old_embeds_context = copy.deepcopy(context.render_context.setdefault(
```

Comment:
Is there a reason to make this a `deepcopy` instead of a shallow `copy`? I understand that `list.copy` is not compatible with Python 2, but why not use `copy.copy`?